### PR TITLE
Add nginx recipe.

### PR DESCRIPTION
**Pull request by sodre**

### CODE REVIEW

**Comment by sodre** [comment]
@jakirkham, if you have time can you do a quick review for the NGINX package?

**Comment by sodre** [comment]
@ocefpaf

This recipe is ready for review when you get a chance.

File: recipes/nginx/build.sh [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      --with-pcre-jit \
+      --with-cc-opt="$cc_opt" \
+      --with-ld-opt="$ld_opt" \
+      --prefix="$PREFIX"
+
+      # this is removed for now because libgd apparently needs libwebp, not yet compiled for linux
+
+fi
+
+make -j2
```
Comment (ocefpaf) [inline recipes/nginx/build.sh]: Please use `CPU_COUNT` here instead of assuming `2`.

File: recipes/nginx/build.sh [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+make install
```
Comment (ocefpaf) [inline recipes/nginx/build.sh]: Is there a `make check` we can run here?

File: recipes/nginx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://nginx.org/download/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - pcre-config.patch
```
Comment (ocefpaf) [inline recipes/nginx/meta.yaml]: Can you add a 1-line comment here explaining the patch?

File: recipes/nginx/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  skip: True  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - libgd
+    - libxslt
+    - openssl 1.0.2*
+    - pcre
+    - zlib
```
Comment (ocefpaf) [inline recipes/nginx/meta.yaml]: Please check the pinnings used in conda-forge and apply them here.

See https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/pin_the_slow_way.py#L28-L71

File: recipes/nginx/build.sh (hunk 1 above)
Comment (sodre) [inline recipes/nginx/build.sh]: Good idea!

File: recipes/nginx/build.sh (hunk 2 above)
Comment (sodre) [inline recipes/nginx/build.sh]: Nope, but it has a comprehensive test after it has been compiled, look at the meta.yaml test section.

File: recipes/nginx/meta.yaml (hunk 3 above)
Comment (sodre) [inline recipes/nginx/meta.yaml]: It is to find pcre in $PREFIX instead of /usr/{lib, include}. Comment added.

File: recipes/nginx/meta.yaml (hunk 4 above)
Comment (sodre) [inline recipes/nginx/meta.yaml]: Let me know what you think.

**Pull request merged**